src/components/Dashboard/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,163 @@
+/*eslint-disable*/
+import React from "react";
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import { NavLink } from "react-router-dom";
+// @material-ui/core components
+import { makeStyles } from "@material-ui/core/styles";
+import Drawer from "@material-ui/core/Drawer";
+import Hidden from "@material-ui/core/Hidden";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Icon from "@material-ui/core/Icon";
+// core components
+import AdminNavbarLinks from '../Navbars/AdminNavbarLinks';
+import RTLNavbarLinks from '../Navbars/RTLNavbarLinks';
+
+import styles from '../../../assets/jss/material-dashboard-react/components/sidebarStyle';
+// @ts-ignore
+const useStyles = makeStyles(styles);
+// @ts-ignore
+export default props => {
+  const classes = useStyles();
+  // verifies if routeName is the one active (in browser input)
+ // @ts-ignore
+  const activeRoute = routeName => window.location.href.indexOf(routeName) > -1 ? true : false;
+  const { color, logo, image, logoText, routes } = props;
+  var links = (
+    <List className={classes.list}>
+     // @ts-ignore
+      {routes.map((prop, key) => {
+        var activePro = " ";
+        var listItemClasses;
+        if (prop.path === "/upgrade-to-pro") {
+          activePro = classes.activePro + " ";
+          listItemClasses = classNames({
+            [" " + classes[color]]: true
+          });
+        } else {
+          listItemClasses = classNames({
+            [" " + classes[color]]: activeRoute(prop.layout + prop.path)
+          });
+        }
+        const whiteFontClasses = classNames({
+          [" " + classes.whiteFont]: activeRoute(prop.layout + prop.path)
+        });
+        return (
+          <NavLink
+            to={prop.layout + prop.path}
+            className={activePro + classes.item}
+            activeClassName="active"
+            key={key}
+          >
+            <ListItem button className={classes.itemLink + listItemClasses}>
+              {typeof prop.icon === "string" ? (
+                <Icon
+                  className={classNames(classes.itemIcon, whiteFontClasses, {
+                    [classes.itemIconRTL]: props.rtlActive
```

Comment:
I wonder what the square brackets are doing here? if this setting a tuple to a tuple or something?